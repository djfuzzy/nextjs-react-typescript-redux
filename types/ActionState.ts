export enum ActionState {
  Initial = 'Initial',
  Processing = 'Processing',
  Completed = 'Completed',
  Failed = 'Failed',
}
