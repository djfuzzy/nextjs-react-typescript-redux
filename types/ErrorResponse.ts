export type ErrorResponse = {
  title: string;
  content: string;
};

export const isErrorResponse = (
  error: ErrorResponse | string | undefined
): error is ErrorResponse => (error as ErrorResponse)?.title !== undefined;
