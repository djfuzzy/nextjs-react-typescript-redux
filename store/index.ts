import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';

import commentReducer from './slices/comments';
import { ErrorResponse } from '../types/ErrorResponse';

const rootReducer = combineReducers({
  comments: commentReducer,
});

export type ErrorResult = {
  error: ErrorResponse | string;
};

export type ThunkConfig = { rejectValue: ErrorResult };

export type CoreState = ReturnType<typeof rootReducer>;

export default configureStore({
  reducer: rootReducer,
});
