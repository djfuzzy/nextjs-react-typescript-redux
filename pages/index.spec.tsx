import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import IndexPage from '.';
import { CoreState } from '../store';
import { ActionState } from '../types/ActionState';
import { Comment } from '../types/Comment';

describe('Index page', () => {
  const mockStore = configureStore<CoreState>();

  // TODO: Move this elsewhere
  const mockComments = [
    {
      id: '1',
      name: 'Homer Simpson',
      content: 'Purple is a fruit.',
      createdAt: new Date('2021-03-22T14:52:10Z'),
    },
  ] as Comment[];

  const entities = mockComments.reduce(
    (entities, entity) => ({ ...entities, [entity.id]: entity }),
    {}
  );

  const store = mockStore({
    comments: {
      entities,
      ids: Object.keys(entities),
      loadState: ActionState.Completed,
      saveState: ActionState.Initial,
      selectedCommentId: undefined,
    },
  });

  it('should match the snapshot', () => {
    const index = (
      <Provider store={store}>
        <IndexPage />
      </Provider>
    );
    const tree = renderer.create(index).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
