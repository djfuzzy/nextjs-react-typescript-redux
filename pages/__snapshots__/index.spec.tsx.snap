// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match the snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <h1
      className="title has-text-centered"
    >
      Next.js Demo
    </h1>
    <p
      className="has-text-centered"
    >
      With:
    </p>
    <ul
      className="has-text-centered"
    >
      <li>
        <strong>
          React
        </strong>
      </li>
      <li>
        <strong>
          Typescript
        </strong>
      </li>
      <li>
        <strong>
          Redux Toolkit
        </strong>
      </li>
      <li>
        <strong>
          Firebase
        </strong>
      </li>
      <li>
        <strong>
          Bulma
        </strong>
      </li>
      <li>
        and more...
      </li>
    </ul>
    <div
      className="comment-form"
    >
      <div
        className="card my-3"
      >
        <header
          className="card-header"
        >
          <p
            className="card-header-title"
          >
            Add new comment
          </p>
        </header>
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="field"
              >
                <label
                  className="label"
                  htmlFor="name"
                >
                  Enter your name
                </label>
                <div
                  className="control"
                >
                  <input
                    autoComplete="name"
                    className="input  "
                    disabled={false}
                    maxLength={100}
                    name="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                  />
                </div>
              </div>
              <div
                className="field"
              >
                <label
                  className="label"
                  htmlFor="content"
                >
                  Enter a comment
                </label>
                <div
                  className="control"
                >
                  <textarea
                    className="textarea "
                    disabled={false}
                    maxLength={500}
                    name="content"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="e.g. These pretzils are making me thirsty."
                  />
                </div>
              </div>
              <div
                className="field is-grouped"
              >
                <div
                  className="control"
                >
                  <button
                    className="button is-link "
                    disabled={false}
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="comment-list"
    >
      <div
        className="card my-3"
      >
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            <table
              className="table is-hoverable "
            >
              <thead>
                <tr>
                  <th>
                    Added
                  </th>
                  <th>
                    Name
                  </th>
                  <th>
                    Comment
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  className="is-clickable"
                  onClick={[Function]}
                >
                  <td>
                    03/22/2021
                  </td>
                  <td>
                    Homer Simpson
                  </td>
                  <td>
                    Purple is a fruit.
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
